hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -447,8 +537,7 @@ private boolean validate(final byte[] b, final int off, final int len)
     Preconditions.checkNotNull(b);
     LOG.debug("read one block requested b.length = {} off {} len {}", b.length,
         off, len);
-
-    if (this.available() == 0) {
+    if (fileStatusInformationPresent.get() && this.available() == 0) {
```

Comment:
Since we are already checking file status info is available here the if check ( !fileStatusInformationPresent.get()) in available() method will never get hit.

Reply:
available() is a public api and can be called from other places as well. check in available is for the fileStatus creation heuristic in available().